### PR TITLE
Add support for max_value to reset the sequence once the max is reached.

### PR DESCRIPTION
**Pull request by hvdklauw**
We need to generator incremental number based on a identifier, this seemed the best solution, but because the numbers are used as a reference they should reset at a certain point.

Thus I added a max_value keyword only argument to allow for this.

Also added tests.

Please merge and release a new version so we can use it.

### CODE REVIEW

**Comment by aaugustin** [comment]
If I read the code correctly, this generates values up to `max_value` inclusive while the usual convention is exclusive — `n` isn't in `range(n)`. Is that correct? Do you mind if I change that part of the behavior?

**Comment by hvdklauw** [comment]
max_value does imply it's inclusive, it's the maximum value you can get back.

Otherwise the kwarg would be reset_at or something I guess.

But feel free, I just need a way to loop the value back to the initial at a certain point.

File: sequences/__init__.py [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
-        *, nowait=False, using=None):
+        *, nowait=False, using=None, max_value=None):
     """
     Return the next value for a given sequence.
 
     """
     # Inner import because models cannot be imported before their application.
     from .models import Sequence
+    if max_value:
+        assert initial_value < max_value
```
Comment (aaugustin) [inline sequences/__init__.py]: With `initial_value = 1` and `max_value = 0`, this assertion doesn't raise an error, while it should.

File: sequences/tests.py [hunk 2]
```diff
@@ -295,3 +303,37 @@ def two(output):
         ]
 
         self.assertSequence(one, two, expected)
+
+    def test_max_value_commit(self):
```
Comment (aaugustin) [inline sequences/tests.py]: Since the two sequences are independent, I don't believe this test validates a behavior that isn't already tested elsewhere.

**Comment by aaugustin** [comment]
@hvdklauw I made a few adjustments and added docs. I'm comfortable with the result. Please confirm that you are as well, then I'll merge and make a release. Thanks!

**Comment by hvdklauw** [comment]
Yes, thanks for the review, I'm happy with the result.

**Comment by aaugustin** [comment]
New version is on PyPI. Enjoy and thanks for your contribution!

**Pull request merged**